misc/python/materialize/cli/run.py

```diff
@@ -158,41 +165,54 @@ def main() -> int:
 
     env = dict(os.environ)
     if args.program in KNOWN_PROGRAMS:
-        (build_retcode, built_programs) = _build(args, extra_programs=[args.program])
+        if args.bazel:
+            build_func = _bazel_build
+        else:
+            build_func = _cargo_build
+
+        (build_retcode, built_programs) = build_func(
+            args, extra_programs=[args.program]
+        )
+
         if args.build_only:
             return build_retcode
 
-        if args.release:
-            if args.sanitizer != "none":
-                artifact_path = MZ_ROOT / "target" / SANITIZER_TARGET / "release"
-            else:
-                artifact_path = MZ_ROOT / "target" / "release"
-        else:
-            if args.sanitizer != "none":
-                artifact_path = MZ_ROOT / "target" / SANITIZER_TARGET / "debug"
-            else:
-                artifact_path = MZ_ROOT / "target" / "debug"
-
-        if args.disable_mac_codesigning:
-            if sys.platform != "darwin":
-                print("Ignoring --disable-mac-codesigning since we're not on macOS")
-            else:
-                print("Disabled macOS Codesigning")
-        elif sys.platform == "darwin":
+        if args.enable_mac_codesigning:
             for program in built_programs:
-                path = artifact_path / program
-                _macos_codesign(str(path))
+                if sys.platform == "darwin":
+                    _macos_codesign(program)
+            if sys.platform != "darwin":
+                print("Ignoring --enable-mac-codesigning since we're not on macOS")
+        else:
+            print("Disabled macOS Codesigning")
 
         if args.wrapper:
             command = shlex.split(args.wrapper)
         else:
             command = []
-        command.append(str(artifact_path / args.program))
+
+        mzdata = MZ_ROOT / "mzdata"
```

Comment:
I'm not convinced this is a good place to put the binaries. mzdata historically only contained runtime data, and so was an easy directory to remove to reset an environment (this is perhaps not true today, but it once was). If that's still the case for mzdata (and thus folks may have like `rf -rf mzdata` somewhere in their local test scripts), somewhere else may be better.

Reply:
Totally makes sense! What do you think about introducing a `.gitignore`-d `build` directory at the top level of the repo and placing them there?

Reply:
SGTM!

Reply:
Ended up calling the new dir `mzbuild`, seemed to line up better with `mzdata`